barretenberg/cpp/src/barretenberg/flavor/goblin_ultra.hpp

```diff
@@ -539,6 +541,8 @@ class GoblinUltra {
 
     class VerifierCommitments : public AllEntities<Commitment, CommitmentHandle> {
       public:
+        static std::vector<std::vector<CommitmentHandle>> get_concatenation_groups() { return {}; }; // WORKTODO
```

Comment:
WORKTODO: adding these when they're not used is a hack, added this to remind myself to look for a better way, maybe a `requires`. Do you see a way to improve or do you think just tech debt this?

Reply:
Turns out that moving the extraction of unshifted, shifted etc to outside of ZM made it easy to get rid of these since now they are only called from within the translator on input to prove and verify. Could add a concept but as long as no one trues to instantiate the Translator with say the Ultra flavor there's no issue.

Reply:
More accurately - they WILL only be called from the translator. Those calls don't exist in this branch but with these updates we're no longer instantiating any classes that need to call them with flavors that don't implement them.